src/verifier/verifier.controller.ts

```diff
@@ -9,14 +10,43 @@ import { VerifierService } from './verifier.service';
 @ApiTags('verifier')
 export class VerifierController {
 
+    private rate_limit_list: { ip: string, count: number }[] = [];
```

Comment:
This seems like a good candidate for a Map, a Record, or a plain object. Especially for that search operation on line 29, it would help a lot with the time complexity of the checkAndLogIp method.